R/api_data.R

```diff
@@ -0,0 +1,119 @@
+#' @title Read data
+#' @description Read data directly into a global environment
+#' @param data character, name of data object
+#' @param filetype character, extension type of the data. Default: "rds"
+#' @param use_cache boolean, use the data stored in cache? Default: FALSE
+#' @return data.frame
+#' @details
+#'   Currently only loading of rds files is supported
+#' @examples
+#'  read_data("fev_data")
+#'  read_data("bcva_data")
+#' @rdname data_api
+#' @export
+#' @importFrom tools file_path_sans_ext
+read_data <- function(data, filetype = "rds", use_cache = FALSE) {
+
+  if(use_cache) {
+    cache_dir <- cache_dir()
+    if(dir.exists(cache_dir)) {
+      items <- tools::file_path_sans_ext(list.files(cache_dir))
+      if(data %in% items){
+        data_path <- file.path(cache_dir, sprintf("%s.%s", data, filetype))
+        message(sprintf("Retrieving %s from cache", data))
+        if(filetype == "rds"){
+          data_out <- readRDS(data_path)
+        }
+      }else{
```

Comment:
good to run styler across the package 
```suggestion
      } else {
```